packages/react-examples/src/react/DetailsList/DetailsList.KeyboardAccessibleResizeAndReorder.Example.tsx

```diff
@@ -0,0 +1,320 @@
+    showDialog();
+  };
+
+  const reorderColumn = (column: IColumn) => {
+    columnToEdit.current = column;
+    clickHandler.current = REORDER;
+    showDialog();
+  };
+
+  const confirmDialog = () => {
+    const detailsList = detailsListRef.current;
+
+    if (textfieldRef.current) {
+      input.current = Number(textfieldRef.current.value);
+    }
+
+    if (columnToEdit.current && input.current && detailsList) {
+      if (clickHandler.current === RESIZE) {
+        const width = input.current;
+        detailsList.updateColumn(columnToEdit.current, { width: width });
+      } else if (clickHandler.current === REORDER) {
+        const targetIndex = selection.mode ? input.current + 1 : input.current;
+        detailsList.updateColumn(columnToEdit.current, { newColumnIndex: targetIndex });
+      }
+    }
+
+    input.current = null;
+    hideDialog();
+  };
+
+  const onColumnClick = (ev: React.MouseEvent<HTMLElement>, column: IColumn): void => {
+    if (column.columnActionsMode !== ColumnActionsMode.disabled) {
+      setContextualMenuProps(getContextualMenuProps(ev, column));
+    }
+  };
+
+  const getContextualMenuProps = (ev: React.MouseEvent<HTMLElement>, column: IColumn): IContextualMenuProps => {
+    const items = [
+      { key: 'resize', text: 'Resize', onClick: () => resizeColumn(column) },
+      { key: 'reorder', text: 'Reorder', onClick: () => reorderColumn(column) },
+      { key: 'sort', text: 'Sort', onClick: () => sortColumn(column) },
+    ];
+
+    return {
+      items: items,
+      target: ev.currentTarget as HTMLElement,
+      gapSpace: 10,
+      isBeakVisible: true,
+      onDismiss: onHideContextualMenu,
+    };
+  };
+
+  const hideDialog = () => setIsDialogHidden(true);
+
+  const showDialog = () => setIsDialogHidden(false);
+
+  const sortColumn = (column: IColumn): void => {
+    let isSortedDescending = column.isSortedDescending;
+
+    // If we've sorted this column, flip it.
+    if (column.isSorted) {
+      isSortedDescending = !isSortedDescending;
+    }
+
+    // Sort the items.
+    const newSortedItems = copyAndSort(sortedItems, column.fieldName!, isSortedDescending);
+
+    columns.map(col => {
+      col.isSorted = col.key === column.key;
+
+      if (col.isSorted) {
+        col.isSortedDescending = isSortedDescending;
+      }
+
+      return col;
+    });
+
+    setSortedItems(newSortedItems);
+  };
+
+  const onHideContextualMenu = React.useCallback(() => setContextualMenuProps(undefined), []);
+
+  const [draggedItem, setDraggedItem] = React.useState(undefined);
+  const [selection] = React.useState(new Selection());
+  const [draggedIndex, setDraggedIndex] = React.useState(-1);
+  const [items, setItems] = React.useState<IExampleItem[]>(createListItems(5, 0));
+  const [sortedItems, setSortedItems] = React.useState<IExampleItem[]>(items);
+  const [columns, setColumns] = React.useState<IColumn[]>(buildColumns(items, true, onColumnClick, undefined, false));
```

Comment:
if you make the columns have `columnActionsMode: ColumnActionsMode.hasDropdown`, they'll have `aria-haspopup="true"` added, and also have a dropdown icon.

The first I think is desirable, and I don't dislike having the icon. I'll leave it up to you, though -- I wouldn't say `aria-haspopup` is required, since without the icon it's not like sighted users have any indication of the menu popup either, so SR users aren't missing out.

Reply:
Made `buildColumn` function accept a new `columnActionMode` argument (4e2870b) to make adding this easier and utilized that in example (13dfd1a)